packages/app/src/routers/ViewRouter/ViewRouter.tsx

```diff
@@ -36,17 +37,28 @@ const useThemeWatcher = () => {
   return theme;
 };
 export interface ViewProps extends RouteComponentProps {
+  accessToken?: string;
   projectId?: string;
   iModelId?: string;
+  versionId?: string;
 }
 const View = (props: ViewProps) => {
   (window as any).ITWIN_VIEWER_HOME = window.location.origin;
   const config = useConfig();
   const buddiRegion = config.buddi?.region
     ? parseInt(config.buddi.region)
     : undefined;
+  const {
+    results: { namedVersion: fetchedVersion },
+  } = useApiData<{ namedVersion: { changesetId: string } }>({
+    accessToken: props.versionId ? props.accessToken : undefined,
+    url: `https://api.bentley.com/imodels/${props.iModelId}/namedversions/${props.versionId}`,
+  });
+
+  const changesetId = props.versionId ? fetchedVersion?.changesetId : undefined;
```

Comment:
I want to keep the meaning so I'll leave it as is, we care about the fetchedVersion.changesetId only if we are provided a versionId, otherwise it is forced to undefined.